src/frontend/js/modules/session.js

```diff
@@ -93,17 +93,30 @@
   })
 
   // TODO (esn-frontend-common-libs#51): Write tests for the new changes (https://github.com/OpenPaaS-Suite/esn-frontend-common-libs/pull/48)
-  .controller('sessionInitESNController', function($scope, esnTemplate, sessionFactory) {
+  .controller('sessionInitESNController', function($scope, $log, esnTemplate, sessionFactory, esnAuth) {
     $scope.session = {
       template: esnTemplate.templates.loading
     };
 
-    sessionFactory.bootstrapSession()
-      .then(() => {
-        // we $apply because otherwise sometimes angular does not detect the change
-        $scope.$apply(() => {
-          $scope.session.template = esnTemplate.templates.success;
-        });
+    esnAuth.init()
+      .then(user => {
+        if (!user) {
+          return esnAuth.signin().catch(err => {
```

Comment:
You should also add this pull request to the `TODO` comment above.

Reply:
I will prefer that we create issue which links to PR with the right label. It will be easier to track

Reply:
Yes, I think we can also do that.